src/CoreWCF.Kafka/src/CoreWCF/Channels/KafkaTransportSecurity.cs

```diff
@@ -0,0 +1,132 @@
+            {
+                throw new ArgumentOutOfRangeException(nameof(value));
+            }
+
+            _credentialType = value;
+        }
+    }
+
+    internal BindingElement CreateBindingElement()
+    {
+        KafkaSecurityBindingElement bindingElement = new();
+
+        if (_mode == KafkaSecurityMode.Transport)
+        {
+            if (_credentialType == KafkaCredentialType.None)
+            {
+                bindingElement.SecurityProtocol ??= SecurityProtocol.Ssl;
+                bindingElement.SslCaPem ??= CaPem;
+            }
+
+            if (_credentialType == KafkaCredentialType.Mtls)
+            {
+                bindingElement.SecurityProtocol ??= SecurityProtocol.Ssl;
+                bindingElement.SslCaPem ??= CaPem;
+                if (ClientCertificate != null)
+                {
+                    bindingElement.SslKeyPem ??= ClientCertificate.KeyPem;
+                    bindingElement.SslKeyPassword ??= ClientCertificate.KeyPassword;
+                    bindingElement.SslCertificatePem ??= ClientCertificate.CertificatePem;
+                }
+            }
+
+            if (_credentialType == KafkaCredentialType.Sasl)
+            {
+                bindingElement.SslCaPem ??= CaPem;
+                bindingElement.SaslMechanism ??= SaslMechanism.Plain;
+                bindingElement.SecurityProtocol ??= SecurityProtocol.SaslSsl;
+                if (UsernamePassword != null)
+                {
+                    bindingElement.SaslUsername ??= UsernamePassword.Username;
+                    bindingElement.SaslPassword ??= UsernamePassword.Password;
+                }
+            }
+        }
+
+        if (_mode == KafkaSecurityMode.TransportCredentialOnly)
+        {
+            bindingElement.SecurityProtocol ??= SecurityProtocol.SaslPlaintext;
+            bindingElement.SaslMechanism ??= SaslMechanism.Plain;
+            if (UsernamePassword != null)
+            {
+                bindingElement.SaslUsername ??= UsernamePassword.Username;
+                bindingElement.SaslPassword ??= UsernamePassword.Password;
+            }
+        }
+
+        if (_mode == KafkaSecurityMode.None)
+        {
+            bindingElement.SecurityProtocol ??= SecurityProtocol.Plaintext;
+        }
+
+        return bindingElement;
+    }
+
+    public string CaPem { get; set; }
+
+    public UsernamePasswordCredential UsernamePassword { get; set; }
+
+    public ClientCertificateCredential ClientCertificate { get; set; }
+}
+
+public class UsernamePasswordCredential
+{
+    public UsernamePasswordCredential(string username, string password)
+    {
+        Username = username;
+        Password = password;
+    }
+
+    public string Username { get; }
+    public string Password { get; }
+}
+
+public class ClientCertificateCredential
+{
+    public string CertificatePem { get; set; }
+    public string KeyPem { get; set; }
+    public string KeyPassword { get; set; }
```

Comment:
it would be nice to keep name same as ConsumerClient..

Reply:
@birojnayak I renamed these properties so they match the ones from Confluent.Kafka